ui/src/app/components/rename-modal.tsx

```diff
@@ -55,6 +54,87 @@ export class RenameModal extends React.Component<Props, States> {
     };
   }
 
+  getFileNameWithoutExtension(name: string) {
+    const lastDotIndex = name?.lastIndexOf('.');
+    return lastDotIndex === -1 ? name : name?.slice(0, lastDotIndex);
+  }
+
+  getFileExtension(name: string) {
+    const lastDotIndex = name?.lastIndexOf('.');
+    return lastDotIndex === -1 ? '' : name?.slice(lastDotIndex);
+  }
+
+  validateNewName(
+    existingNames: string[],
+    oldName: string,
+    newName: string,
+    resourceType: ResourceType
+  ) {
+    const { lowerCaseNames, newNameNoExtension } =
+      this.getFilteredListOfFileNamesWithoutExtension(
+        existingNames,
+        oldName,
+        newName
+      );
+
+    const errors = validate(
```

Comment:
optional: could skip the creation of `errors` and do `return validate(...` instead